clang/test/dpct/cmake_migration/case_004/expected.txt

```diff
@@ -0,0 +1,18 @@
+target_link_sycl_libraries(main PRIVATE cxxopts fmt::fmt -lcublas)
+
+target_link_sycl_libraries(transformer_engine PUBLIC 
+   CUDA::cublas
+   CUDA::cuda_driver
+   CUDA::cudart
+   CUDA::nvrtc
+   CUDA::nvToolsExt
+   CUDNN::cudnn)
```

Comment:
After a further thought, target_link_libraries() is not a cuda specific cmake syntax, so  target_link_sycl_libraries is not needed introduced, we can cuda libraries to sycl mappings if the mapping is available, or "" if mapping is not available.

Reply:
Some libraries need to be linked based on windows specific or linux specific names, for this reason we had discussed using a helper function. Do you think it is not required anymore?

Reply:
@daiyaan-ahmed6  synced with @zhimingwang36:
target_link_libraries() is not a cuda specific cmake syntax, so target_link_sycl_libraries is not needed introduced here.
#1. We map cuda library to sycl mapping if the mapping is available, or "" if mapping is not available.
#2. Some libraries need to be linked based on windows specific or linux specific names, we import a variable in helper function. Take "CUDNN::cudnn" for example, we introduce variable "dnnlib", and set its value to "dnnl.lib" on Windows, and to "dnnl" on Linux, then migrate target_link_sycl_libraries(transformer_engine PUBLIC CUDNN::cudnn) to target_link_sycl_libraries(transformer_engine PUBLIC dnnlib).

Reply:
Done.